### ISSUE TITLE
Issue : Embedded view doesn't work if master pk column has another name than "id"

### ISSUE THREAD

**Issue by Chlipouni** [saltcorn/saltcorn#728]
It is possible to define a table with a primary key column that has a specific name (other than "id").

A view can be defined on this table and works correctly (url filter can be applied on the column name).
But when this view is embedded in another one, its content remains empty.

By renaming the primary key column to "id", the embedded view works fine again.

**Comment by AleksandrSokolov** [comment]
@Chlipouni Hello, could you provide test exampe - to repeat your sitation.
You can create it in https://uat_reproducing_errors.saltcorn.com/ or somelse.

**Comment by Chlipouni** [comment]
@AleksandrSokolov,
My account is created on https://uat_reproducing_errors.saltcorn.com/
Can you give me the necessary privileges to create new tables and views ?

**Comment by AleksandrSokolov** [comment]
@Chlipouni I gave you admin rights.

**Comment by Chlipouni** [comment]
@AleksandrSokolov ,

When I try to create a new table, the "id" column is automatically created and I can't rename it.

To reproduce the test case, it is necessary to create the following tables directly in the PostgreSQL database :
`create table uid_master (
  uid integer generated by default as identity not null, 
  lib text,
  constraint pk_uid_master primary key (uid)
);

create table uid_detail (
  uid integer generated by default as identity not null,
  lib text,
  uid_master integer,
  constraint pk_uid_detail primary key (uid),
  constraint fk_uid_detail_uid_master foreign key (uid_master) references uid_master
);

insert into uid_master (lib) values ('Master 1');
insert into uid_master (lib) values ('Master 2');

insert into uid_detail (lib, uid_master) values ('Master 1 - Detail 1', 1);
insert into uid_detail (lib, uid_master) values ('Master 2 - Detail 1', 2);
insert into uid_detail (lib, uid_master) values ('Master 2 - Detail 2', 2);`

**Comment by AleksandrSokolov** [comment]
@Chlipouni I'll try at tomorrow everning/weekend

**Comment by AleksandrSokolov** [comment]
@Chlipouni I made few changes for non id primary key tables. Please check - is the problem persist

**Comment by Chlipouni** [comment]
@AleksandrSokolov,

Just tried with 0.5.1-alpha.3 and the problem is not resolved.
I can create the test case on  https://uat_reproducing_errors.saltcorn.com/ but you have to create the two tables first directly in the PostgreSQL database.

**Comment by AleksandrSokolov** [comment]
@glutamate Tom, Please create tables in https://uat_reproducing_errors.saltcorn.com/ - we need to reproduce problem
create table uid_master (
uid integer generated by default as identity not null,
lib text,
constraint pk_uid_master primary key (uid)
);

create table uid_detail (
uid integer generated by default as identity not null,
lib text,
uid_master integer,
constraint pk_uid_detail primary key (uid),
constraint fk_uid_detail_uid_master foreign key (uid_master) references uid_master
);

insert into uid_master (lib) values ('Master 1');
insert into uid_master (lib) values ('Master 2');

insert into uid_detail (lib, uid_master) values ('Master 1 - Detail 1', 1);
insert into uid_detail (lib, uid_master) values ('Master 2 - Detail 1', 2);
insert into uid_detail (lib, uid_master) values ('Master 2 - Detail 2', 2);

**Comment by glutamate** [comment]
@AleksandrSokolov we want these in the `uat_reproducing_errors` schema i think - I have run in psql:

```
create table uat_reproducing_errors.uid_master (
uid integer generated by default as identity not null,
lib text,
constraint pk_uid_master primary key (uid)
);

create table uat_reproducing_errors.uid_detail (
uid integer generated by default as identity not null,
lib text,
uid_master integer,
constraint pk_uid_detail primary key (uid),
constraint fk_uid_detail_uid_master foreign key (uid_master) references uat_reproducing_errors.uid_master
);

insert into uat_reproducing_errors.uid_master (lib) values ('Master 1');
insert into uat_reproducing_errors.uid_master (lib) values ('Master 2');

insert into uat_reproducing_errors.uid_detail (lib, uid_master) values ('Master 1 - Detail 1', 1);
insert into uat_reproducing_errors.uid_detail (lib, uid_master) values ('Master 2 - Detail 1', 2);
insert into uat_reproducing_errors.uid_detail (lib, uid_master) values ('Master 2 - Detail 2', 2);
```

**Comment by AleksandrSokolov** [comment]
@glutamate yes

**Comment by Chlipouni** [comment]
I have just created two views :
- UID Detail List :      List of uid_detail records
- UID Master Show : On record from uid_master with the UID Detail List as an embedded view

UID Detail List works correctly with the following url : /view/UID Detail List?uid_master=2

But when I request the UID Master Show with the following url : /view/UID Master Show?uid=2,
only the master record is displayed, the embedded view stays empty.

**Comment by glutamate** [comment]
fixed in #770

**Comment by Chlipouni** [comment]
Thank you for the fix
